src/Nethermind/Nethermind.Consensus/Processing/OneTimeProcessor.cs

```diff
@@ -14,13 +15,13 @@ public class OneTimeChainProcessor : IBlockchainProcessor
         public ITracerBag Tracers => _processor.Tracers;
 
         private readonly IBlockchainProcessor _processor;
-        private readonly IReadOnlyDbProvider _readOnlyDbProvider;
+        private readonly Action _clearTemporaryState;
 
         private object _lock = new();
 
-        public OneTimeChainProcessor(IReadOnlyDbProvider readOnlyDbProvider, IBlockchainProcessor processor)
+        public OneTimeChainProcessor(Action clearTemporaryState, IBlockchainProcessor processor)
```

Comment:
It's sketchy indeed. I was thinking of removing this and adding a `Reset` to the IWorldState or something instead of relying on this. Therefore I don't want to put too much effort into this, so I'd just use `Action` instead. Besides, it's more of a reset instead of a dispose.